packages/aws-cdk-lib/aws-stepfunctions/lib/states/succeed.ts

```diff
@@ -54,6 +61,7 @@ export class Succeed extends State {
   public toStateJson(): object {
     return {
       Type: StateType.SUCCEED,
+      StateName: this.stateName,
```

Comment:
If we put `StateName` in this return object, what ultimately happens to it?

Have you seen the documentation for [Amazon States Language](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html)?

This is what we are targeting. It looks like this:

```ts
{
  "StartAt": "<StateName>",
  "States": {
    "<StateName1>": {
      "Type": "<Whatever>",
      <Properties>
    },
    "<StateName2>": {
      "Type": "<MoreWhatever>",
    }
}
```

That is ultimately what we want to change.